hera_sim/interpolators.py

```diff
@@ -131,3 +157,106 @@ def _check_pol(self, pol):
 
+# XXX can we think of a better name for this? it's a bit unfortunate that
+# XXX interp1d is already taken...
+class Hera1dInterp:
+    """
+    This class provides an interface for creating either a numpy.poly1d or a 
+    scipy.interpolate.interp1d interpolation object from a reference file. This 
+    class is intended to be used primarily as a helper function for various 
+    functions in the hera_sim repository.
+    """
+    def __init__(self, datafile, **interp_kwargs):
+        self.datafile = _check_path(datafile)
+        self._data = np.load(datafile)
+        self._obj = interp_kwargs.pop("object", None)
+        self._interp_type = interp_kwargs.pop("interpolator", "poly1d")
+        self._interp_kwargs = interp_kwargs
+        self._check_format()
+    
+    # TODO: update docstring
+    """
+    Initialize a BeamSize object from a given reference file and choice of 
+    interpolator.
+
+    Args:
+        ref_file (str):
+            Absolute path to the reference file to be used for making the
+            BeamSize interpolator. This must be either a .npy or .npz file,
+            depending on which interpolation method is chosen.
+
+        interpolator (str):
+            Choice of interpolation object to be used. Must be either 'poly1d' 
+            or 'interp1d'. If 'poly1d' is chosen, then ref_file must be a 
+            .npy file. If 'interp1d' is chosen, then ref_file must be a .npz 
+            file with two arrays stored in its archive: 'freqs' and 'beam'.
+
+        **interp_kwargs (dict, optional):
+            Keyword arguments to be passed to the interpolator in the case that
+            'interp1d' is chosen.
+
+    Raises:
+        AssertionError:
+            This is raised if the choice of interpolator and the required type
+            of the ref_file do not agree (i.e. trying to make a 'poly1d' object
+            using a .npz file as a reference). An AssertionError is also raised
+            if the .npz for generating an 'interp1d' object does not have the
+            correct arrays in its archive.
+    """
+
+    def __call__(self, freqs):
+        return self._interpolator(freqs)
+
+    # XXX see note in Tsky class for same method
+    @staticmethod
+    def _names():
+        return ("omega_p", "bandpass", )
+
+    @cached_property
+    def _interpolator(self):
+        if self._interp_type=='poly1d':
+            return np.poly1d(self._data)
+        else:
+            # if not using poly1d, then need to get some parameters for
+            # making the interp1d object
+            obj = self._data[self._obj]
+            freqs = self._data['freqs']
+
+            # use a cubic spline by default, but override this if the user
+            # specifies a different kind of interpolator
+            kind = self._interp_kwargs.pop('kind', 'cubic')
+            return interp1d(freqs, obj, kind=kind, **self._interp_kwargs)
+
+    def _check_format(self):
+        if self._obj is None:
```

Comment:
Instead of supporting different kinds of objects, why not just make a base class and subclasses? That's what they exist for! Then the user can easily define `my_bandpass = Bandpass(...)`.